apps/web/src/pages/activities/components/ActivityGraph.tsx

```diff
@@ -0,0 +1,349 @@
+import { useQuery } from 'react-query';
+import styled from '@emotion/styled';
+import {
+  BarElement,
+  CategoryScale,
+  Chart as ChartJS,
+  Legend,
+  LinearScale,
+  ScriptableContext,
+  Title,
+  Tooltip,
+} from 'chart.js';
+import { Bar } from 'react-chartjs-2';
+import moment from 'moment';
+import { useEffect, useState } from 'react';
+import { getActivityGraphStats } from '../../../api/activity';
+import { colors, Text } from '../../../design-system';
+import { IActivityGraphStats, IChartData } from '../interfaces';
+import { activityGraphStatsMock } from '../consts';
+
+ChartJS.register(CategoryScale, LinearScale, BarElement, Title, Tooltip, Legend);
+
+function checkIsTriggerSent(activityGraphStats: IActivityGraphStats[] | undefined) {
+  return activityGraphStats?.length && activityGraphStats?.length > 0;
+}
+
+export function ActivityGraph() {
+  const [isTriggerSent, setIsTriggerSent] = useState<boolean>(false);
+  const { data: activityGraphStats, isLoading: loadingActivityStats } = useQuery<IActivityGraphStats[]>(
+    'activityGraphStats',
+    getActivityGraphStats
+  );
+
+  useEffect(() => {
+    if (checkIsTriggerSent(activityGraphStats)) {
+      setIsTriggerSent(true);
+    }
+  }, [activityGraphStats]);
+
+  return (
+    <Wrapper>
+      {!isTriggerSent ? (
```

Comment:
```suggestion
      {!isTriggerSent && ! loadingActivityStats ? (
```

this will prevent a flashing state while loading